### DISCUSSION TITLE
README out of date?

### DISCUSSION THREAD

**Discussion by grubstarstar** [socketio/socket.io#3816] [Q&A]
It looks like the readme is out of date with the latest API interface. E.g. when setting up with express. The new one exports a Server object which isn't mentioned in the README.

**Comment by grubstarstar** [comment]
I was previously doing this to integrate it with express, but can't work out how to do it with the latest interface after upgrading.

    import express from 'express'
    import io from 'socket.io'

    const app = express()
    const server = http.createServer(app)
    const socket = io(server)

I tried:

    import express from 'express'
    import { Server } from 'socket.io'

    const app = express()
    const server = http.createServer(app)
    const socket = new Server(server)
    
but my client sockets are failing to connect now.

**Comment by darrachequesne** [comment]
As long as you don't use `app.listen()` (hinted [here](https://socket.io/docs/v3/server-initialization/#With-Express)), the following code should work:

```js
import express from 'express'
import { Server } from 'socket.io'

const app = express()
const server = http.createServer(app)
const socket = new Server(server)

server.listen(8080);
```

**Reply by grubstarstar** [reply]
Thanks for the quick response. Yeah I'm doing `server.listen(<port>)` already. Do I have to use a specific version of the client to match the version of the server?

Just having difficulty because it seems like the documentation hasn't been updated for the new APIs in both `socket.io-client` and `socket.io`. Unless they're hiding somewhere?!

Also the TypeScript types don't seem to match up to what I'm actually getting. E.g. `import { Socket, Manager } from 'socket.io-client'` for the client gives me a class Socket type but it's undefined.

I feel like I'm missing something.
